### PR TITLE
#155903085 Define the API Endpoints

### PR DESCRIPTION
**Pull request by wycliffkas**
#### What does this PR do?
- defines the API Endpoints
#### Description of Task to be completed?
- defines the API Endpoints
- using special retrofit annotations (@GET) to encode details about the parameters and request method.
#### What are the relevant pivotal tracker stories?
[#155903085](https://www.pivotaltracker.com/story/show/155903085)

### CODE REVIEW

**Comment by huxaiphaer** [comment]
The build is failing here.

**Comment by wycliffkas** [comment]
> The build is failing here.

let me work on it

File: .idea/checkstyle-idea.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        <entry key="copy-libs" value="false" />
+        <entry key="location-0" value="BUNDLED:(bundled):Sun Checks" />
+        <entry key="location-1" value="BUNDLED:(bundled):Google Checks" />
+        <entry key="scan-before-checkin" value="false" />
+        <entry key="scanscope" value="JavaOnly" />
+        <entry key="suppress-errors" value="false" />
+      </map>
+    </option>
+  </component>
+</project>
```
Comment (huxaiphaer) [inline .idea/checkstyle-idea.xml]: Why are the `.idea `files still there? kindly remove all of them.

File: .idea/codeStyles/Project.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        <option name="com.jetbrains.cidr.lang.util.OCDeclarationKind" value="InstanceMethod" />
+        <option name="com.jetbrains.cidr.lang.util.OCDeclarationKind" value="DeallocMethod" />
+      </class>
+      <extensions>
+        <pair source="cpp" header="h" fileNamingConvention="NONE" />
+        <pair source="c" header="h" fileNamingConvention="NONE" />
+      </extensions>
+    </Objective-C-extensions>
+  </code_scheme>
+</component>
```
Comment (huxaiphaer) [inline .idea/codeStyles/Project.xml]: Remove the  `.idea `folder.

**Review by huxaiphaer — COMMENTED** [review]
Kindly act on the feedback

File: .idea/codeStyles/Project.xml (hunk 2 above)
Comment (wycliffkas) [reply .idea/codeStyles/Project.xml]: I have removed it, it had remained in the cache

**Comment by huxaiphaer** [comment]
LGTM

**Pull request merged**